packages/bitcore-node/test/integration/models/block.integration.ts

```diff
@@ -0,0 +1,611 @@
+        value: 500.0,
+        address: 'mkjB6LmjiNfJWgH4aP4v1GkFjRcQTfDSfj',
+        spentTxid: 'eec8570a0c960b19fa6c86c71a06ebda379b86b5fe0be0e64ba83b2e0a3d05a3',
+        spentHeight: 9
+      });
+
+      const localTip = await BlockModel.getLocalTip({ chain: 'BTC', network: 'regtest' });
+      expect(localTip).to.exist;
+      expect(localTip.chain).to.equal('BTC');
+      expect(localTip.height).to.equal(7);
+      expect(localTip.network).to.equal('regtest');
+      expect(localTip.version).to.equal(536870912);
+      expect(localTip.hash).to.equal('3279069d22ce5af68ef38332d5b40e79e1964b154d466e7fa233015a34c27312');
+      expect(localTip.merkleRoot).to.equal('8c29860888b915715878b21ce14707a17b43f6c51dfb62a1e736e35bc5d8093f');
+      expect(localTip.previousBlockHash).to.equal('2a883ff89c7d6e9302bb4a4634cd580319a4fd59d69e979b344972b0ba042b86');
+      expect(localTip.nonce).to.equal(3);
+      expect(localTip.size).to.equal(264);
+      expect(localTip.bits).to.equal(545259519);
+      expect(localTip.processed).to.equal(true);
+
+      await BlockModel.handleReorg({
+        header: {
+          prevHash: '12c719927ce18f9a61d7c5a7af08d3110cacfa43671aa700956c3c05ed38bdaa',
+          hash: '4c6872bf45ecab2fb8b38c8b8f50fc4a8309c6171d28d479b8226afcb1a99920',
+          time: 1526326785,
+          version: '536870912',
+          merkleRoot: '8c29860888b915715878b21ce14707a17b43f6c51dfb62a1e736e35bc5d8093f',
+          bits: parseInt('207fffff', 16).toString(),
+          nonce: '3'
+        },
+        chain: 'BTC',
+        network: 'regtest'
+      });
+
+      const blocks = await BlockModel.find({
+        chain: 'BTC',
+        network: 'regtest'
+      }).exec();
+      expect(blocks.length).to.equal(2);
+
+      const removedBlock = await BlockModel.find({
+        chain: 'BTC',
+        network: 'regtest',
+        hash: '3279069d22ce5af68ef38332d5b40e79e1964b154d466e7fa233015a34c27312',
+        height: {
+          $gte: localTip.height
+        }
+      }).exec();
+      expect(removedBlock).to.be.an('array').that.is.empty;
+
+      const transaction = await TransactionModel.find({
+        chain: 'BTC',
+        network: 'regtest'
+      }).exec();
+      expect(transaction.length).to.equal(1);
+
+      const removedTransaction = await TransactionModel.find({
+        chain: 'BTC',
+        network: 'regtest',
+        blockHeight: {
+          $gte: localTip.height
+        }
+      }).exec();
+      expect(removedTransaction).to.be.an('array').that.is.empty;
+
+      const coinModel = await CoinModel.find({
+        chain: 'BTC',
+        network: 'regtest',
+      }).exec();
+      expect(coinModel.length).to.equal(1);
+
+      const removedCoin = await CoinModel.find({
+        chain: 'BTC',
+        network: 'regtest',
+        mintHeight: {
+          $gte: localTip.height
+        }
+      });
+      expect(removedCoin).to.be.an('array').that.is.empty;
+
+      const updateCoin = await CoinModel.find({
```

Comment:
maybe call this `unspentCoins` or something similar

Reply:
resolved

Reply:
confirmed